src/datasets/utils/tf_utils.py

```diff
@@ -166,16 +173,18 @@ def dataset_to_tf(
         collate_fn=collate_fn,
         collate_fn_args=collate_fn_args,
         columns_to_np_types=columns_to_np_types,
-        return_dict=False,  # TF expects numpy_function to return a list and will not accept a dict
+        return_dict=False,
     )
 
+    # This works because dictionaries always output in the same order
+    tout = [tf.dtypes.as_dtype(dtype) for dtype in columns_to_np_types.values()]
+
     @tf.function(input_signature=[tf.TensorSpec(None, tf.int64)])
     def fetch_function(indices):
-        output = tf.numpy_function(
+        output = tf.py_function(
```

Comment:
The change from `numpy_function` to `py_function` here should be okay, but is it necessary? I understand you want to be able to pass a single index instead of a batch of indices, but I feel like you could just pass a Numpy scalar in that case (with shape == ()) and leave the rest of the code the same.

Reply:
More information at https://github.com/tensorflow/tensorflow/blob/388d952114e59a1aeda440ed4737b29f8b7c6e8a/tensorflow/python/ops/script_ops.py#L210

Reply:
Ah, that makes sense! Okay, I'm fine with this change.